curated_transformers/tokenization/bbpe_tokenizer.py

```diff
@@ -22,22 +27,30 @@ def __init__(
     def _decode(self, input: Iterable[Iterable[int]]) -> List[str]:
         return [self.processor.decode_from_ids(ids) for ids in input]
 
-    def _encode(self, input: Iterable[str]) -> PiecesWithIds:
+    def _encode(self, input: Iterable[MergedInputChunks]) -> PiecesWithIds:
         ids = []
         pieces = []
 
-        for text in input:
-            text_ids = []
-            text_pieces = []
-
-            for idx, token in enumerate(text.split(" ")):
-                if idx != 0:
-                    token = " " + token
-                token_ids, token_pieces = self.processor.encode(token)
-                text_ids.extend(token_ids)
-                text_pieces.extend(token_pieces)
-
-            ids.append(text_ids)
-            pieces.append(text_pieces)
+        for seq in input:
+            seq_ids = []
+            seq_pieces = []
+
+            for chunk in seq:
+                if isinstance(chunk, MergedSpecialPieceChunk):
```

Comment:
Do we want to add validate the iterable's elements' type here (and in other tokenizers) as well?

Reply:
Ah, right. That's true.